lib/components/form/batch-settings.tsx

```diff
@@ -136,81 +146,60 @@ class BatchSettings extends Component<Props, State> {
-
-  render() {
-    const { activeSearch, config, currentQuery, intl, modeOptions } = this.props
-    const { expanded, selectedModes } = this.state
-    return (
-      <div role="group">
-        <fieldset>
-          <ModeButtonsFullWidthContainer>
-            <legend>
-              {intl.formatMessage({
-                id: 'components.BatchSettings.modeSelector'
-              })}
-            </legend>
-            <ModeButtonsFullWidth
-              className="flex"
-              modeOptions={modeOptions}
-              onClick={this._onClickMode}
-              selectedModes={selectedModes}
-            />
-          </ModeButtonsFullWidthContainer>
-        </fieldset>
-        <MainSettingsRow>
-          <SettingsPreview
-            aria-controls="batch-preferences"
-            aria-expanded={expanded === 'SETTINGS'}
-            aria-label={intl.formatMessage({
-              id: 'components.BatchSettings.settings'
-            })}
-            expanded={expanded === 'SETTINGS'}
-            onClick={this._toggleSettings}
-          >
-            {coreUtils.query.isNotDefaultQuery(currentQuery, config) && (
-              <Dot className="dot" />
-            )}
-            <StyledIconWrapper size="2x">
-              <Cog />
-            </StyledIconWrapper>
-          </SettingsPreview>
-          {expanded === 'SETTINGS' && (
-            <BatchPreferencesContainer id="batch-preferences">
-              <BatchPreferences />
-            </BatchPreferencesContainer>
-          )}
-
-          <StyledDateTimePreviewContainer
-            aria-controls="date-time-modal"
-            aria-expanded={expanded === 'DATE_TIME'}
-            aria-label="Date/Time settings"
-            expanded={expanded === 'DATE_TIME'}
-            onClick={this._toggleDateTime}
-          >
-            <StyledDateTimePreview hideButton />
-          </StyledDateTimePreviewContainer>
-          {expanded === 'DATE_TIME' && (
-            <DateTimeModalContainer id="date-time-modal">
-              <DateTimeModal />
-            </DateTimeModalContainer>
-          )}
+  const _toggleModeButton = useCallback(
+    (buttonId: string, newState: boolean) => {
+      let newButtons
+      if (newState) {
+        newButtons = [...enabledModeButtons, buttonId]
+      } else {
+        newButtons = enabledModeButtons.filter((c) => c !== buttonId)
+      }
 
+      // encodeQueryParams serializes the mode buttons for the URL
+      // It uses encodeQueryParams to get nice looking URL params and consistency
+      setUrlSearch(
+        encodeQueryParams(queryParamConfig, { modeButtons: newButtons })
+      )
+    },
+    [enabledModeButtons, setUrlSearch]
+  )
+
+  // We can rely on this existing, as there is a default
+  const baseColor = getComputedStyle(document.documentElement).getPropertyValue(
+    '--main-base-color'
+  )
+  const accentColor = tinycolor(baseColor).darken(10)
+
+  return (
+    <div role="group">
```

Comment:
Remove the "group" role per client request, so that the `<fieldset>` element can be correctly announced by assistive technology.

Reply:
Right, this had been mentioned to me, thanks for the reminder. I replaced the div with a fragment.